src/main/java/org/opentripplanner/api/resource/PlanGenerator.java

```diff
@@ -1097,28 +1091,32 @@ private static WalkStep createWalkStep(Graph graph, State s) {
         if (en instanceof AreaEdge) {
             step.area = true;
         } else if (en instanceof StreetEdge) {
-                StreetEdge street = (StreetEdge) en;
-                step.benches += street.getBenchCount();
-                step.toilets += street.getToiletCount();
-
-                // NIH Properties
-                OptionSet nihOptions = street.getExtraOptionFields();
-                if (nihOptions != null) {
-                    step.lastAudited = new Date();
-                    OptionAttribute rest = nihOptions.getOption(NihOption.REST);
-                    if ((rest != null) && (rest != Rest.NONE_AVAILABLE)) {
-                        step.rest = rest.getLabel();
-                    }
-                    OptionAttribute aesthetics = nihOptions.getOption(NihOption.AESTHETIC);
-                    if (aesthetics == Aesthetics.YES) {
-                        step.aesthetics = true;
-                    }
-                    OptionAttribute uneven = nihOptions.getOption(NihOption.XSLOPE);
-                    if (uneven == XSlope.SLOPED) {
-                        step.unevenSurfaces = true;
-                    }
+            StreetEdge street = (StreetEdge) en;
+            step.benches += street.getBenchCount();
+            step.toilets += street.getToiletCount();
+            float maxSlope = Math.abs(street.getMaxSlope());
+            if (maxSlope > step.maxSlope) {
```

Comment:
In case there are multiple edges in this step, it will return the max across them all.